client/reader/search-stream/suggestion-provider.jsx

```diff
@@ -23,32 +24,64 @@ function suggestionsFromTags( count, tags ) {
 		if ( tags.length <= count ) {
 			return [];
 		}
-		return map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+		return map(
+			sampleSize( tags, count ),
+			( tag, i ) => {
+				const text = ( tag.displayName || tag.slug ).replace( /-/g, ' ' );
+				const ui_algo = 'read:search-suggestions:tags/1';
+				return suggestionWithRailcar( text, ui_algo, i );
+			}
+		);
 	}
 	return null;
 }
 
 function suggestionsFromPicks( count ) {
 	const lang = i18nUtils.getLocaleSlug().split( '-' )[ 0 ];
-
 	if ( suggestions[ lang ] ) {
-		return sampleSize( suggestions[ lang ], count );
+		return map(
+			sampleSize( suggestions[ lang ], count ),
+			( tag, i ) => {
+				const ui_algo = 'read:search-suggestions:picks/1';
+				return suggestionWithRailcar( tag, ui_algo, i );
+			}
+		);
 	}
 	return null;
 }
 
+function suggestionWithRailcar( text, ui_algo, position ) {
+	return {
+		text: text,
+		railcar: {
+			railcar: analytics.tracks.createRandomId() + '-' + position,
+			ui_algo: ui_algo,
+			ui_position: position,
+			rec_result: text,
+		}
+	};
+}
+
 function getSuggestions( count, tags ) {
 	const tagSuggestions = suggestionsFromTags( count, tags );
+
+	// return null to suppress showing any suggestions until tag subscriptions load.
 	if ( tagSuggestions === null ) {
-		// return null to supperess showing any suggestions until tag subscriptions load
 		return null;
 	}
 
-	if ( tagSuggestions.length ) {
-		return tagSuggestions;
-	}
+	const newSuggestions = !! tagSuggestions.length
+		? tagSuggestions
+		: suggestionsFromPicks( count );
+
+	trackSuggestionRailcarRender( newSuggestions );
```

Comment:
this gets called during a render (the connects `mapStateToProps`) which is a time that you aren't supposed to dispatch during.  I'll add a fix